src/SFML/Audio/SoundFileFactory.cpp

```diff
@@ -154,19 +125,45 @@ std::unique_ptr<SoundFileReader> SoundFileFactory::createReaderFromStream(InputS
 ////////////////////////////////////////////////////////////
 std::unique_ptr<SoundFileWriter> SoundFileFactory::createWriterFromFilename(const std::filesystem::path& filename)
 {
-    // Register the built-in readers/writers on first call
-    ensureDefaultReadersWritersRegistered();
-
     // Test the filename in all the registered factories
-    for (const WriterFactory& writerFactory : s_writers)
+    for (const auto& [fpCreate, fpCheck] : getWriterFactoryMap())
     {
-        if (writerFactory.check(filename))
-            return writerFactory.create();
+        if (fpCheck(filename))
+            return fpCreate();
     }
 
     // No suitable writer found
     err() << "Failed to open sound file (format not supported)\n" << formatDebugPathInfo(filename) << std::endl;
     return nullptr;
 }
 
+
+////////////////////////////////////////////////////////////
+SoundFileFactory::ReaderFactoryMap& SoundFileFactory::getReaderFactoryMap()
+{
+    using namespace priv;
```

Comment:
I'd personally keep it explicit, so it's clear what we're pulling from the `priv` namespace

Reply:
Done